optimum/torch_export_patches/_core.py

```diff
@@ -0,0 +1,527 @@
+            ``_PATCHES_``, ``_PATCHED_CLASS_``
+        rewrite: list of methods to automatically rewrite
+            before exporting, methods with control flow need to be rewritten
+            before being exported if the execution path depends on the inputs,
+            this is done by function :func:`transform_method
+            <onnx_diagnostic.torch_export_patches.patch_module.transform_method>`,
+            its documentation provides possible values
+        dump_rewriting: dumps rewriting information in file beginning with that prefix
+        verbose: to show which patches is applied
+
+    The list of available patches:
+
+    * ``torch.jit.isinstance``
+    * ``torch._dynamo.mark_static_address``
+    * ``torch._subclasses.fake_impls.infer_size``
+    * ``torch.vmap``
+    * fix missing method ``name`` for ``sympy.S.IntegerConstant``
+    * ``AttentionMaskConverter._make_causal_mask``
+    * Serialization of ``MambaCache`` (in :epkg:`transformers`)
+    * Serialization of ``DynamicCache`` (in :epkg:`transformers`)
+    * reduce errors due to shape inference
+    * fixes some transformers classes,
+      see :mod:`onnx_diagnostic.torch_export_patches.patches.patch_transformers`
+
+    Serialization issues happen when a module takes one input or output
+    has a type :func:`torch.export.export` cannot serialize.
+
+    Examples:
+        .. code-block:: python
+
+            with torch_export_patches(patch_transformers=True) as modificator:
+                inputs = modificator(inputs)
+                onx = to_onnx(..., inputs, ...)
+
+        .. code-block:: python
+
+            with torch_export_patches(patch_transformers=True) as modificator:
+                inputs = modificator(inputs)
+                onx = torch.onnx.export(..., inputs, ...)
+
+        It can be used as well to fix the torch export:
+
+        .. code-block:: python
+
+            with torch_export_patches(patch_transformers=True) as modificator:
+                inputs = modificator(inputs)
+                ep = torch.export.export(..., inputs, ...)
+
+        When running the model through the exported program, only the
+        serialization functions need to be restored:
+
+        .. code-block:: python
+
+            with register_additional_serialization_functions() as modificator:
+                inputs = modificator(inputs)
+                ep = torch.export.export(..., inputs, ...)
+
+        When exporting a model with a cache, the following error message
+        may appear ``AssertionError: Mutating module attribute _seen_tokens during export.``.
+        It can be avoided by setting ``strict=False`` when call :func:`torch.export.export`.
+    """
+    if rewrite:
+        raise ValueError(
+            "The argument `rewrite` is not supported at the moment, as the automatic rewriting is not implemented yet."
+        )
+        # from .patch_module import torch_export_rewrite
+
+        # with torch_export_rewrite(
```

Comment:
@xadupre I suggest we disable dum_rewriting for now (It's control flow auto-rewriting?). We can have the core functionality first. Let me know if you think other ways.

Reply:
Done